py/server/deephaven/column.py

```diff
@@ -32,47 +32,122 @@ def __repr__(self):
-
-
-@dataclass
-class InputColumn(Column):
-    """ An InputColumn represents a user defined column with some input data. """
-    input_data: Any = field(default=None)
-
-    def __post_init__(self):
+    def j_object(self) -> jpy.JType:
+        return self.j_column_definition
+
+    @cached_property
+    def name(self) -> str:
+        return self.j_column_definition.getName()
+
+    @cached_property
+    def data_type(self) -> DType:
+        return from_jtype(self.j_column_definition.getDataType())
+
+    @cached_property
+    def component_type(self) -> DType:
+        return from_jtype(self.j_column_definition.getComponentType())
+
+    @cached_property
+    def column_type(self) -> ColumnType:
+        return ColumnType(self.j_column_definition.getColumnType())
+
+
+class Column(ColumnDefinition):
+    """A Column object represents a column definition in a Deephaven Table."""
+
+    def __init__(
+            self,
+            name: str,
+            data_type: DType,
+            component_type: DType = None,
+            column_type: ColumnType = ColumnType.NORMAL,
+    ):
+        super().__init__(
+            ColumnDefinition.of(
+                name, data_type, component_type, column_type
+            ).j_column_definition
+        )
+
+
+class InputColumn:
+    """An InputColumn represents a user defined column with some input data."""
+
+    def __init__(
+            self,
+            name: str = None,
+            data_type: DType = None,
+            component_type: DType = None,
+            column_type: ColumnType = ColumnType.NORMAL,
+            input_data: Any = None,
+    ):
         try:
-            if self.input_data is None:
-                self.j_column = _JColumn.empty(self.j_column_header)
-            else:
-                if self.data_type.is_primitive:
-                    self.j_column = _JColumn.ofUnsafe(self.name, dtypes.array(self.data_type, self.input_data,
-                                                                              remap=dtypes.null_remap(self.data_type)))
-                else:
-                    self.j_column = _JColumn.of(self.j_column_header, dtypes.array(self.data_type, self.input_data))
+            self.column_definition = ColumnDefinition.of(
+                name, data_type, component_type, column_type
+            )
+            self.j_column = self._to_j_column(input_data)
         except Exception as e:
             raise DHError(e, f"failed to create an InputColumn ({self.name}).") from e
 
+    @property
+    def name(self):
+        return self.column_definition.name
+
+    @property
+    def data_type(self) -> DType:
+        return self.column_definition.data_type
+
+    @property
+    def component_type(self) -> DType:
+        return self.column_definition.component_type
+
+    @property
+    def column_type(self) -> ColumnType:
+        return self.column_definition.column_type
+
+    def _to_j_column(self, input_data: Any = None) -> jpy.JType:
```

Comment:
pydoc

Reply:
What's our stance on pydoc for private methods?